### PR TITLE
Add Maverick v2 to `dex.trades` spell

### PR DESCRIPTION
**Pull request by gte620v**
### Contribution type
Please check the type of contribution this pull request is for:

- [ ] New spell(s)
- [x] Adding to existing spell lineage
- [ ] Bug fix

**Note:** You can safely discard any section below which doesn't apply based on selection above

---

### For adding to existing spell lineage
If you are adding to an existing spell lineage, please provide the following information:

- **Description:** Adding maverick v2 `dex.trade` spells.  There is only a small difference in the event format of maverick v1.

### CODE REVIEW

File: dex/models/trades/arbitrum/platforms/maverick_v2_arbitrum_base_trades.sql [hunk 1]
```diff
@@ -0,0 +1,21 @@
+{{
+    config(
+        schema = 'maverick_v2_arbitrum',
```
Comment (Hosuke) [inline dex/models/trades/arbitrum/platforms/maverick_v2_arbitrum_base_trades.sql:3]: We may need to add the model schema for these new spells.

**Review by Hosuke — CHANGES_REQUESTED** [review]
Hi @gte620v, please add the model schema to corresponding `_schema.yml` files. We may add the seed files for testing purpose as well.

File: dex/models/trades/arbitrum/platforms/maverick_v2_arbitrum_base_trades.sql (hunk 1 above)
Comment (Hosuke) [reply dex/models/trades/arbitrum/platforms/maverick_v2_arbitrum_base_trades.sql:3]: Something like this:
https://github.com/duneanalytics/spellbook/blob/main/dex/models/trades/arbitrum/_schema.yml#L722

Comment (gte620v) [reply dex/models/trades/arbitrum/platforms/maverick_v2_arbitrum_base_trades.sql:3]: @Hosuke  Great; I made the changes.

I haven't been able to debug the build failure.  

image

Seems that there is some issue with the seed. But all seems fine in the file.  Any ideas?

Comment (Hosuke) [reply dex/models/trades/arbitrum/platforms/maverick_v2_arbitrum_base_trades.sql:3]: We need to add the schema for seed files as well.

File: dex/seeds/trades/maverick_zksync_base_trades_seed.csv [hunk 2]
```diff
@@ -1,2 +1,3 @@
 blockchain,project,version,block_date,tx_hash,evt_index,token_bought_address,token_sold_address,block_number,token_bought_amount_raw,token_sold_amount_raw
 zksync,maverick,1,2023-10-23,0xbe2700aac1174728513863a4f42d37aa9ac7a7d73389fa00a5abcfbd3570d5a6,4,0x2039bb4116B4EFc145Ec4f0e2eA75012D6C0f181,0x3355df6D4c9C3035724Fd0e3914dE96A5a83aaf4,17114057,1723727835619188444,1723200
+zksync,maverick,2,2024-07-24,0xceb5de2540e7df1ba059a6da0c8a81632736725c54a0c39fb1e62b216cadbad2,88,0x5aea5775959fbc2557cc8789bc1bf90a239d9a91,0x5a7d6b2f92c77fad6ccabd7ee0624e64907eaf3e,39925150,25855263746109189,505494599746163070374
```
Comment (Hosuke) [inline dex/seeds/trades/maverick_zksync_base_trades_seed.csv:3]: Please verify this seed record.
https://dune.com/queries/3941353

**Review by Hosuke — CHANGES_REQUESTED** [review]

File: dex/seeds/trades/maverick_zksync_base_trades_seed.csv (hunk 2 above)
Comment (gte620v) [reply dex/seeds/trades/maverick_zksync_base_trades_seed.csv:3]: It seems the zksync factory contract was misnamed (I submitted a renaming request to `V2Factory` a few days ago, but it is still in the queue.  

Using the existing name, I can run this in dune and i get the expected result:

```
SELECT
  t.evt_block_time AS block_time,
  t.evt_block_number AS block_number,
  t.recipient AS taker,
  TRY_CAST(NULL AS VARBINARY) AS maker,
  t.amountOut AS token_bought_amount_raw,
  t.amountIn AS token_sold_amount_raw,
  CASE
    WHEN TRY_CAST(JSON_EXTRACT_SCALAR(t.params, '$.tokenAIn') AS BOOLEAN) = TRUE
    THEN f.tokenB
    ELSE f.tokenA
  END AS token_bought_address,
  CASE
    WHEN TRY_CAST(JSON_EXTRACT_SCALAR(t.params, '$.tokenAIn') AS BOOLEAN) = TRUE
    THEN f.tokenA
    ELSE f.tokenB
  END AS token_sold_address,
  t.contract_address AS project_contract_address,
  t.evt_tx_hash AS tx_hash,
  t.evt_index
FROM maverick_v2_zksync.V2Pool_evt_PoolSwap AS t
INNER JOIN maverick_v2_zksync.MaverickV2Factory_evt_PoolCreated AS f
  ON t.contract_address = f.poolAddress
WHERE
  t.evt_tx_hash = 0xceb5de2540e7df1ba059a6da0c8a81632736725c54a0c39fb1e62b216cadbad2
  ```
  
  But the CI can't seem to find `maverick_v2_zksync.MaverickV2Factory_evt_PoolCreated`.  any ideas, @Hosuke ?

Comment (Hosuke) [reply dex/seeds/trades/maverick_zksync_base_trades_seed.csv:3]: We may need to add the new source table into the corresponding`sources.yml`.

File: sources/_sector/dex/trades/zksync/_sources.yml [hunk 3]
```diff
@@ -5,6 +5,10 @@ sources:
     tables:
       - name: pool_evt_Swap
       - name: factory_evt_PoolCreated
+  - name: maverick_v2_zksync
+    tables:
+      - name: V2Pool_evt_PoolSwap
+      - name: V2Factory_evt_PoolCreated
```
Comment (Hosuke) [inline sources/_sector/dex/trades/zksync/_sources.yml:11]:
```suggestion
      - name: V2Pool_evt_PoolSwap
      - name: V2Factory_evt_PoolCreated
      - name: MaverickV2Factory_evt_PoolCreated
```

**Review by Hosuke — APPROVED** [review]
LGTM.✅
Thank you @gte620v

**Review by jeff-dude — APPROVED** [review]

**Comment by jeff-dude** [comment]
we are blocked by merge conflicts, without access to help resolve

**Comment by gte620v** [comment]
> we are blocked by merge conflicts, without access to help resolve

try now

**Comment by jeff-dude** [comment]
thank you, let's see CI pass to be safe

**Pull request merged**